node/src/proxy_server/mod.rs

```diff
@@ -362,8 +369,12 @@ impl ProxyServer {
     }
 
     fn handle_dns_resolve_failure(&mut self, msg: &ExpiredCoresPackage<DnsResolveFailure_0v1>) {
+        let return_route_id = match self.rri_from_remaining_route(&msg.remaining_route) {
+            Some(rri) => rri,
+            None => return, // TODO: Eventually we'll have to do something better here, but we'll probably need some heuristics.
+        };
```

Comment:
`get_return_route_info()` has been modified to accept only the `return_route_id` (a `u32`) rather than the whole `Route`; so here we're pulling the `return_route_id` out of the `Route` we have.